### PR TITLE
feat(coderd): insert provisioner daemons to the database again

### PR DESCRIPTION
**Pull request by johnstcn**
Refers to https://github.com/coder/coder/issues/10676

- Add UpsertProvisionerDaemon
- Remove unused InsertProvisionerDaemon
- Update existing references
- Add ProvisionerDaemonUser RBAC resource
- Insert provisioner daemons again
- Update unit tests

### CODE REVIEW

File: coderd/database/dbauthz/dbauthz.go [hunk 1]
```diff
@@ -226,6 +227,7 @@ var (
 					rbac.ResourceOrganization.Type:       {rbac.ActionCreate},
 					rbac.ResourceOrganizationMember.Type: {rbac.ActionCreate},
 					rbac.ResourceOrgRoleAssignment.Type:  {rbac.ActionCreate},
+					rbac.ResourceProvisionerDaemon.Type:  {rbac.ActionCreate, rbac.ActionUpdate, rbac.ActionDelete},
```
Comment (johnstcn) [inline coderd/database/dbauthz/dbauthz.go:230]: review: for PSK auth

File: coderd/database/dbauthz/dbauthz.go [hunk 2]
```diff
@@ -3069,6 +3063,17 @@ func (q *querier) UpsertOAuthSigningKey(ctx context.Context, value string) error
 	return q.db.UpsertOAuthSigningKey(ctx, value)
 }
 
+func (q *querier) UpsertProvisionerDaemon(ctx context.Context, arg database.UpsertProvisionerDaemonParams) (database.ProvisionerDaemon, error) {
+	res := rbac.ResourceProvisionerDaemon
+	if arg.Tags[provisionersdk.TagScope] == provisionersdk.ScopeUser {
+		res = rbac.ResourceProvisionerDaemonUser.WithOwner(arg.Tags[provisionersdk.TagOwner])
+	}
```
Comment (johnstcn) [inline coderd/database/dbauthz/dbauthz.go:3070]: review: technically duplicated from `ProvisionerDaemon.RBACObject()` but not sure it's time yet to refactor this.

**Comment by spikecurtis** [comment]
So, let's talk about `last_seen_at` --- what does "seen" mean here?

In this PR, we set it when the provisioner first connects to coderd, and I don't believe we update it.  That's not going to work with auto-deleting provisioners after `last_seen_at` is more than 7 days ago.

You mention in your comments wanting to bump `last_seen_at` whenever a provisioner updates a job. That doesn't really make sense to me since you might have specialized provisioners that only rarely are issued a job.

**Comment by mtojek** [comment]
> In this PR, we set it when the provisioner first connects to coderd, and I don't believe we update it

I haven't reviewed the PR yet, but the idea was to add an extra "ping" call, so that provisioners can report their status.

**Pull request closed without merging**